db/users-schema.sql

```diff
@@ -10,4 +10,56 @@ CREATE TABLE Users (
   createdAt datetime,
   updatedAt datetime,
   PRIMARY KEY (id)
-);
\ No newline at end of file
+);
+
+<<<<<<< HEAD
+INSERT INTO Users (email, password, createdAt, updatedAt)
+VALUES ("test@test.com", "test", '1901-01-01', '1901-01-01'),
+("john@doe.com", "test", '1901-01-01', '1901-01-01');
+
+CREATE TABLE Reviews (
+  id INT NOT NULL AUTO_INCREMENT,
+  title VARCHAR(100) NOT NULL,
+  description VARCHAR(300) NOT NULL,
+  createdAt datetime,
+  updatedAt datetime,
+  hostId INT,
+  UserId INT,
+  PRIMARY KEY (id),
+  FOREIGN KEY (hostId)
+  REFERENCES Users (id) ON DELETE SET NULL ON UPDATE CASCADE,
+  FOREIGN KEY (UserId)
+  REFERENCES Users (id) ON DELETE CASCADE ON UPDATE CASCADE
+);
+
+CREATE TABLE UserReviews (
+  id INT NOT NULL AUTO_INCREMENT,
+  createdAt datetime,
+  updatedAt datetime,
+  ReviewId INTEGER, 
+  UserId INTEGER,
+  PRIMARY KEY (id), 
+  FOREIGN KEY (ReviewId)  
+  REFERENCES Reviews (id) ON DELETE CASCADE ON UPDATE CASCADE, 
+  FOREIGN KEY (UserId) 
+  REFERENCES Users (id) ON DELETE CASCADE ON UPDATE CASCADE
+);
+=======
+CREATE TABLE shows(
+id INT NOT NULL AUTO_INCREMENT,
+showsID INT NOT NULL,
+showsName VARCHAR (100) NOT NULL,
+showsPlot VARCHAR (150) NOT NULL,
+showsYear INT (4) NOT NULL,
+PRIMARY KEY (id)
+);
+
+CREATE TABLE posts(
+id INT NOT NULL AUTO_INCREMENT,
+postsID INT NOT NULL,
+posts VARCHAR (150) NOT NULL,
+showsID INT NOT NULL,
+PRIMARY KEY (id)
+);
+
+>>>>>>> 5f54ac4f951bbb04fd09052122003fb3b840c087
```

Comment:
This right here is a merge conflict. There were overlapping changes on two different branches. When they were merged together git wasn't sure how to resolve it and needs help in how the final merge should look. 

There are two segments of code git struggled to merge. One is from lines `15-47` and the other is from line `47-56`. Conflicts are denoted by the use of `>>>>`, `<<<<`, and `=====`. If they aren't addressed they often break your app to inform you of a problem. 

The merge conflict is likely causing an error and throwing the user back to the homepage as a default behavior. 

Address this to see if it resolves your issue. This would be a solid first step in determining what the problem is.

Reply:
You can read more about merge conflicts here :smiley: 

https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-merge-conflicts

Reply:
From the looks of it there are multiple files with merge conflicts. 
```
db/shows-schema.sql
db/users-schema.sql
models/review.js
public/home.html 
```

Reply:
original: Error: Incorrect integer value: '' for column 'hostId' at row 1
      at Packet.asError (/Users/lindseytummond/Desktop/UofA Bootcamp/Homework/bingeflix/node_modules/mysql2/lib/packets/packet.js:712:17)
      at Execute.execute (/Users/lindseytummond/Desktop/UofA Bootcamp/Homework/bingeflix/node_modules/mysql2/lib/commands/command.js:28:26)
      at Connection.handlePacket (/Users/lindseytummond/Desktop/UofA Bootcamp/Homework/bingeflix/node_modules/mysql2/lib/connection.js:417:32)
      at PacketParser.onPacket (/Users/lindseytummond/Desktop/UofA Bootcamp/Homework/bingeflix/node_modules/mysql2/lib/connection.js:75:12)
      at PacketParser.executeStart (/Users/lindseytummond/Desktop/UofA Bootcamp/Homework/bingeflix/node_modules/mysql2/lib/packet_parser.js:75:16)
      at Socket.<anonymous> (/Users/lindseytummond/Desktop/UofA Bootcamp/Homework/bingeflix/node_modules/mysql2/lib/connection.js:82:25)
      at Socket.emit (events.js:315:20)
      at addChunk (_stream_readable.js:302:12)
      at readableAddChunk (_stream_readable.js:278:9)
      at Socket.Readable.push (_stream_readable.js:217:10)
      at TCP.onStreamRead (internal/stream_base_commons.js:186:23) {
    code: 'ER_TRUNCATED_WRONG_VALUE_FOR_FIELD',
    errno: 1366,
    sqlState: 'HY000',
    sqlMessage: "Incorrect integer value: '' for column 'hostId' at row 1",
    sql: 'INSERT INTO `Reviews` (`id`,`title`,`description`,`createdAt`,`updatedAt`,`hostId`) VALUES (DEFAULT,?,?,?,?,?);',
    parameters: [
      'ozark',
      'awesome',
      '2020-05-29 16:19:47',
      '2020-05-29 16:19:47',
      ''
    ]
  },
  sql: 'INSERT INTO `Reviews` (`id`,`title`,`description`,`createdAt`,`updatedAt`,`hostId`) VALUES (DEFAULT,?,?,?,?,?);',
  parameters: [
    'ozark',
    'awesome',
    '2020-05-29 16:19:47',
    '2020-05-29 16:19:47',
    ''
  ]
}

Reply:
the database appears to be structured correctly. The issue seems to be that when the POST request payload is sent upon creating a review, the line referencing the foreign key of hostId is null and the database is unable to handle it.